src/components/Sidebar/CalendarPanel.vue

```diff
@@ -27,16 +33,64 @@
       end?: Date;
     } = {};
 
+    private defaultDate!: DateItem;
+
     mounted() {
       const { start_date, end_date } = this.$route.query;
 
-      if (start_date && end_date)
+      if (start_date && end_date){
+
         this.dates = {
           start: new Date(start_date as string),
           end: new Date(end_date as string),
         };
+      }
+    }
+
+
+    async fetch(shownMonth = this.defaultDate){
```

Comment:
Could we pass in this data instead of fetching it within the component? That way we could reuse this component in other areas of the website.

Reply:
Yeah we could, except we would probably need to fetch twice from the news page. The trigger for fetching new data on the calendar is when a user switches months, so I'm not sure how we would detect that. If we can't detect the month change then we would need to fetch all the data for all the months in order to fill the calendar.

Reply:
> Yeah we could, except we would probably need to fetch twice from the news page
That seems fine - you can execute them in parallel
> The trigger for fetching new data on the calendar is when a user switches months, so I'm not sure how we would detect that
You would have an event like `@monthChanged` (if the calendar library doesn't expose an event, you can add a watcher to a property via the `@Watch` decorator) and then in the handler you'd 1) remove the current data passed to the calendar (so that it's not stale) and then re-fetch.